pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -43,37 +43,42 @@ const (
 	maxDurationBeforeRetry time.Duration = 2 * time.Minute
 )
 
-// GoRoutineMap defines the supported set of operations.
-type GoRoutineMap interface {
-	// Run adds operationName to the list of running operations and spawns a new
-	// go routine to execute the operation. If an operation with the same name
-	// already exists, an error is returned. Once the operation is complete, the
-	// go routine is terminated and the operationName is removed from the list
-	// of executing operations allowing a new operation to be started with the
-	// same name without error.
-	Run(operationName string, operationFunc func() error) error
+// GoroutineMap defines the supported set of operations.
+type GoroutineMap interface {
+	// Run adds operationName/operationSubName to the list of running operations
```

Comment:
This comment  is good at explaining the behaviour but not the motivation. As commented below we should be able to split the routine into two. Then add a comment just for Sub along the lines of: 

```
/// Here we have to build a sub dependendancy. We want to run the routine only when the concatenation of the target and subTarget are both not being operated on by anything.